pkg/network/node/vxlan_monitor.go

```diff
@@ -0,0 +1,252 @@
+package node
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/origin/pkg/util/ovs"
+)
+
+// egressVXLANMonitor monitors the health of automatic egress IPs by periodically checking
+// "ovs-ofctl dump-flows" output and noticing if the number of packets sent over VXLAN to
+// an egress node has increased, but the number of packets received over VXLAN from the
+// node hasn't. If that happens, it marks the node as being offline, and then starts
+// periodically pinging the node's SDN IP address, until the incoming packet count
+// increases, at which point it marks the node online again.
+//
+// The fact that we (normally) use pod-to-egress traffic to do the monitoring rather than
+// actively pinging the nodes means that if an egress node falls over while no one is
+// using an egress IP on it, we won't notice the problem until someone does try to use it.
+// So, eg, the first pod created in a namespace might spend 5 seconds trying to talk to a
```

Comment:
It's (max) 7 seconds... right?  Or am I missing something.

Every pollInterval (5s) we check the traffic to each node.  If there's a mismatch we go into retry mode which re-checks every 1s and call it failed after 2 retries.  So we will call it bad in best case 2s, worst case 7s, and average 4.5s.

Reply:
Yes. Changed it from "5" to "several"